src/Akka.Persistence.Hosting.Tests/UnifiedApiSpecs.cs

```diff
@@ -0,0 +1,322 @@
+            .Should().Be(typeof(UnifiedApiTestResources.TestReadAdapter).TypeQualifiedName());
+
+        var bindings = journalConfig.GetStringList(
+            $"event-adapter-bindings.\"{typeof(UnifiedApiTestResources.TestEvent).TypeQualifiedName()}\"");
+        bindings.Should().BeEquivalentTo("test-adapter", "test-reader");
+    }
+}
+
+/// <summary>
+/// Test WithSnapshot(SnapshotOptions) without builder
+/// </summary>
+public sealed class SnapshotOptionsOnlySpec : Akka.Hosting.TestKit.TestKit
+{
+    public SnapshotOptionsOnlySpec(ITestOutputHelper output) : base(output: output)
+    {
+    }
+
+    protected override void ConfigureServices(HostBuilderContext context, IServiceCollection services)
+    {
+        base.ConfigureServices(context, services);
+        services.AddHealthChecks();
+    }
+
+    protected override void ConfigureAkka(AkkaConfigurationBuilder builder, IServiceProvider provider)
+    {
+        var options = new UnifiedApiTestResources.TestSnapshotOptions { IsDefaultPlugin = true };
+        builder.WithSnapshot(options);
+    }
+
+    [Fact]
+    public void WithSnapshot_should_configure_snapshot_options_only()
+    {
+        var config = Sys.Settings.Config;
+        config.GetString("akka.persistence.snapshot-store.plugin")
+            .Should().Be("akka.persistence.snapshot-store.test-snapshot");
+        config.HasPath("akka.persistence.snapshot-store.test-snapshot")
+            .Should().BeTrue();
+    }
+}
+
+/// <summary>
+/// Test WithJournalAndSnapshot without builders
+/// </summary>
+public sealed class JournalAndSnapshotWithoutBuildersSpec : Akka.Hosting.TestKit.TestKit
+{
+    public JournalAndSnapshotWithoutBuildersSpec(ITestOutputHelper output) : base(output: output)
+    {
+    }
+
+    protected override void ConfigureServices(HostBuilderContext context, IServiceCollection services)
+    {
+        base.ConfigureServices(context, services);
+        services.AddHealthChecks();
+    }
+
+    protected override void ConfigureAkka(AkkaConfigurationBuilder builder, IServiceProvider provider)
+    {
+        var journalOptions = new UnifiedApiTestResources.TestJournalOptions { IsDefaultPlugin = true };
+        var snapshotOptions = new UnifiedApiTestResources.TestSnapshotOptions { IsDefaultPlugin = true };
+        builder.WithJournalAndSnapshot(journalOptions, snapshotOptions);
+    }
+
+    [Fact]
+    public void WithJournalAndSnapshot_should_configure_both_without_builders()
+    {
+        var config = Sys.Settings.Config;
+        config.GetString("akka.persistence.journal.plugin")
+            .Should().Be("akka.persistence.journal.test-journal");
+        config.GetString("akka.persistence.snapshot-store.plugin")
+            .Should().Be("akka.persistence.snapshot-store.test-snapshot");
+    }
+}
+
+/// <summary>
+/// Test WithJournalAndSnapshot with builder actions
+/// </summary>
+public sealed class JournalAndSnapshotWithBuildersSpec : Akka.Hosting.TestKit.TestKit
```

Comment:
End to end integration test - also asserts that the health checks on each plugin work.